src/app/auth/login.tsx

```diff
@@ -19,45 +25,136 @@ function LoginScreen() {
     router.replace(path);
   };
 
-  const signInWithEmail = async () => {
+  const signIn = async () => {
     setLoading(true);
-    const { error } = await sessionHandler.signInWithEmail(email, password);
+    let email;
+    if (validator.isEmail(emailOrUsername)) {
+      email = emailOrUsername;
+
+      const { count } = await supabase
+        .from('profiles')
+        .select(`*`, { count: 'exact' })
+        .eq('email', email);
+
+      if (count == 0) {
+        setError(
+          'An account with that email does not exist. Please try again.',
+        );
+        return;
+      }
+    } else {
+      const { data, error } = await supabase
+        .from('profiles')
+        .select(`email`)
+        .limit(1)
+        .eq('username', emailOrUsername);
+
+      if (data && data?.length > 0 && !error) {
+        email = data[0].email;
+      } else {
+        setError(
+          'An account with that username does not exist. Please try again.',
+        );
+        setLoading(false);
+        return;
+      }
+    }
 
-    if (error) Alert.alert(error.message);
+    const { error } = await sessionHandler.signInWithEmail(email, password);
+    if (error)
+      setError('The username and/or password is incorrect. Please try again.');
     else resetAndPushToRouter('/home');
     setLoading(false);
   };
 
   return (
-    <View style={globalStyles.auth_container}>
-      <View style={[globalStyles.verticallySpaced, globalStyles.mt20]}>
-        <Input
-          label="Email"
-          leftIcon={{ type: 'font-awesome', name: 'envelope' }}
-          onChangeText={text => setEmail(text)}
-          value={email}
-          placeholder="email@address.com"
-          autoCapitalize="none"
+    <SafeAreaView style={[globalStyles.authContainer, styles.flex]}>
+      <View>
+        <Text style={[globalStyles.h4, styles.title]}>
+          Read stories from young creators
```

Comment:
Looking at the designs, it seems like we want an intentional line break between the words from and young just to make the 2 lines more even